### PR TITLE
Add `installers` stage to the release process

### LINKED ISSUE #340: Add a `subctl` phase in release automation
....after submariner-operator phase.

Part of [Epic: Move subctl to its own repo](https://github.com/submariner-io/enhancements/issues/89#)

### PR DESCRIPTION
**Pull request by mkolesnik**
Add `installers` stage before `released` as now `subctl` lives in its
own repository and hence would be the last thing released.
Hence, the installers (charts and operator) need to be released before
`subctl`.

Resolves #340 

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: scripts/lib/utils [hunk 1]
```diff
@@ -1,15 +1,17 @@
 # shellcheck shell=bash
 # shellcheck disable=SC2034 # We declare some shared variables here
 
-readonly PROJECTS=(admiral cloud-prepare lighthouse shipyard submariner submariner-charts submariner-operator)
-readonly ADMIRAL_CONSUMERS=(cloud-prepare lighthouse submariner submariner-operator)
-readonly SHIPYARD_CONSUMERS=(admiral lighthouse submariner submariner-operator)
+readonly PROJECTS=(admiral cloud-prepare lighthouse shipyard subctl submariner submariner-charts submariner-operator)
+readonly ADMIRAL_CONSUMERS=(cloud-prepare lighthouse subctl submariner submariner-operator)
+readonly SHIPYARD_CONSUMERS=(admiral lighthouse subctl submariner submariner-operator)
 readonly OPERATOR_CONSUMES=(submariner cloud-prepare lighthouse)
```
Comment (tpantelis) [inline scripts/lib/utils]: **submariner-operator** no longer uses **cloud-prepare**.

**Review by tpantelis — APPROVED** [review]

File: scripts/do-release.sh [hunk 2]
```diff
@@ -267,20 +288,9 @@ read_release_file
...
-    release_admiral
-    ;;
-projects)
-    release_projects
-    ;;
-released)
-    release_all
+branch|shipyard|admiral|projects|installers|released)
+    # shellcheck disable=SC2086
+    release_${release['status']}
```
Comment (skitt) [inline scripts/do-release.sh]: This *should* work here:

```suggestion
    "release_${release['status']}"
```

**Review by skitt — CHANGES_REQUESTED** [review]
Minor tweak to avoid a shellcheck override.

**Review by skitt — APPROVED** [review]

**Comment by tpantelis** [comment]
What about building releases of versions prior to the `subctl` repo refactor? It seems we need to keep the prior behavior in that case.

**Comment by mkolesnik** [comment]
> What about building releases of versions prior to the `subctl` repo refactor? It seems we need to keep the prior behavior in that case.

You;re right, but that would be quite problematic with the existing code to support two split build routes.
We should probably create a stable branch for pre-0.13 builds and handle them there (not great but at least would work).

**Comment by tpantelis** [comment]
> > What about building releases of versions prior to the `subctl` repo refactor? It seems we need to keep the prior behavior in that case.
> 
> You;re right, but that would be quite problematic with the existing code to support two split build routes. We should probably create a stable branch for pre-0.13 builds and handle them there (not great but at least would work).

yeah that sounds reasonable. Realistically it will probably be just one more 0.12 release that we'll have to build anyway.

**Comment by tpantelis** [comment]
> > > What about building releases of versions prior to the `subctl` repo refactor? It seems we need to keep the prior behavior in that case.
> > 
> > 
> > You;re right, but that would be quite problematic with the existing code to support two split build routes. We should probably create a stable branch for pre-0.13 builds and handle them there (not great but at least would work).
> 
> yeah that sounds reasonable. Realistically it will probably be just one more 0.12 release that we'll have to build anyway.

OK - I created a [pre-0.13 branch](https://github.com/submariner-io/releases/tree/pre-0.13/) using these commands:

```bash
  git checkout -B pre-0.13 remotes/upstream/devel
  git push upstream pre-0.13
```

If it looks good we should be able to merge this. cc @skitt

**Comment by mkolesnik** [comment]
> OK - I created a [pre-0.13 branch](https://github.com/submariner-io/releases/tree/pre-0.13/) using these commands:
> 
> ```shell
>   git checkout -B pre-0.13 remotes/upstream/devel
>   git push upstream pre-0.13
> ```
> 
> If it looks good we should be able to merge this. cc @skitt

I think we should follow same naming convention, so I created `release-0.12`

I based it on most recent 0.12 release commit, but not including any 0.13 (as it's not relevant for that branch).

I'll merge this one so I can start `0.13.0-m2` and if we need any adjustments we can of course make them later

**Pull request merged**